File: android/app/src/main/java/com/cobaltloom/loraviewer/data/remote/TrailRouteDateFormatter.kt
package com.cobaltloom.loraviewer.data.remote

import java.time.Instant
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter

/** TrailRouteView sends/receives all timestamps as "yyyy-MM-dd HH:mm:ss" in UTC. */
object TrailRouteDateFormatter {
    private val formatter: DateTimeFormatter =
        DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").withZone(ZoneOffset.UTC)

    fun parse(value: String): Instant? =
        try {
            Instant.from(formatter.parse(value))
        } catch (e: Exception) {
            null
        }
}

File: LoraViewer/Models/TrailRouteDateFormatter.swift
import Foundation

/// TrailRouteView sends/receives all timestamps as "yyyy-MM-dd HH:mm:ss" in UTC.
enum TrailRouteDateFormatter {
    static let utc: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd HH:mm:ss"
        f.timeZone = TimeZone(identifier: "UTC")
        f.locale = Locale(identifier: "en_US_POSIX")
        return f
    }()
}
